### ISSUE TITLE
Bad media query for show-for-large-only

### ISSUE THREAD

**Issue by nagylzs** [zurb/foundation-sites#11758]
## What should happen?

Show the element with "show-for-large-only" class for large screen sizes.

## What happens instead?

It is never displayed ( display: none ).

## Possible Solution

A possible (and probably the only correct) solution is to change this media query:

```
 @media screen and (max-width: 63.99875em), screen and (min-width: 75em) {
   .show-for-large-only {
     display: none !important; } }
```

I cannot think of a size that has max-width 63.9 and min-width 75em at the same time. :-D

## Test Case and/or Steps to Reproduce (for bugs)

Just simply create a div with "show-for-large-only" class, and put something inside it. It will never be displayed, regardless of the screen size.

## Context

The biggest problem with the context is that it is not possible to do a correct workaround by writting CSS. Consider this quick fix:

```
.show-for-large-only {
    @media (max-width: 64em) {
        display: none !important;
    }

    @media (min-width: 64em) {
        display: block !important;
    }
}

```
This will change the buggy "display: none" to "display: block". But that is partly bad. Above 64em, the display should be UNCHANGED instead of CHANGED TO BLOCK.

In other words: the only correct way to fix this is to change it in foundation.

**Comment by DanielRuf** [comment]
Hi @nagylzs,

thanks for reporting this.
It seems this is the case for all `-only` breakpoints.

**Comment by DanielRuf** [comment]
https://github.com/zurb/foundation-sites/blob/develop-v6.5/scss/components/_visibility.scss#L57

https://github.com/zurb/foundation-sites/blob/develop-v6.5/scss/util/_breakpoint.scss

@nagylzs do you know or can you tests which version was the last one which does not have this bug? Maybe we can bisect this.

**Comment by DanielRuf** [comment]
https://github.com/zurb/foundation-sites/blob/develop-v6.5/scss/components/_visibility.scss#L21

https://github.com/zurb/foundation-sites/blob/develop-v6.5/scss/util/_breakpoint.scss

@nagylzs do you know or can you tests which version was the last one which does not have this bug? Maybe we can bisect this.

**Comment by DanielRuf** [comment]
https://github.com/zurb/foundation-sites/blob/develop-v6.5/scss/components/_visibility.scss#L40

https://github.com/zurb/foundation-sites/blob/develop-v6.5/scss/util/_breakpoint.scss

@nagylzs do you know or can you tests which version was the last one which does not have this bug? Maybe we can bisect this.

**Comment by DanielRuf** [comment]
I will provide a patch and open a PR for this.

**Comment by DanielRuf** [comment]
https://github.com/zurb/foundation-sites/blob/develop-v6.5/scss/components/_visibility.scss#L40

https://github.com/zurb/foundation-sites/blame/develop-v6.5/scss/components/_visibility.scss#L40

https://github.com/zurb/foundation-sites/commit/f3212070361efdae63f2f823b0544d06f75df3c1

I will provide a patch and open a PR for this.

**Comment by DanielRuf** [comment]
https://github.com/zurb/foundation-sites/blob/develop-v6.5/scss/components/_visibility.scss#L40

https://github.com/zurb/foundation-sites/blame/develop-v6.5/scss/components/_visibility.scss#L40

https://github.com/zurb/foundation-sites/commit/f3212070361efdae63f2f823b0544d06f75df3c1

**Comment by DanielRuf** [comment]
So far this seems correct.
It is shown between 1024px and 1200px.
See https://codepen.io/DanielRuf/pen/arqENo

**Comment by DanielRuf** [comment]
See https://github.com/zurb/foundation-sites/blob/3568b101dbc8eba9911f4a7a3a1b7b52231d6896/scss/settings/_settings.scss#L110-L111

**Comment by nagylzs** [comment]
Mine was 6.5.1

Daniel Ruf <notifications@github.com> ezt írta (időpont: 2019. máj. 23.,
Cs, 14:48):

**Comment by DanielRuf** [comment]
Hi @nagylzs,

Did you check my codepen?

**Comment by nagylzs** [comment]
It seems that there is nothing above large. I was looking for `show-for-large` but I got confused, because it was different in foundation 5. ( show-for-large-up was used in version 5 ).

Then only one question remains: which class should I used for anything that is "bigger than large"? There is no show-for-xlarge-only class defined.

**Comment by DanielRuf** [comment]
Generally it can be configured in the settings.
See https://github.com/zurb/foundation-sites/blob/develop/scss/util/_breakpoint.scss#L36

And large is set as the maximum there.
You can try `show-for-large`. Does this not solve your problem?
https://github.com/zurb/foundation-sites/blob/develop/dist/css/foundation.css#L5389

**Comment by nagylzs** [comment]
Well, I need a class that represents "bigger than large". It cannot be achieved by adding another breakpoint, because that would only change the name of the "biggest named breakpoint". Maybe this makes my point clear:

![size_classes](https://user-images.githubusercontent.com/15814836/58322327-e2c31600-7e20-11e9-9390-0c6243d2670b.png)

**Comment by DanielRuf** [comment]
Right. Well, currently the code has no check if the class list for the breakpoints ends with large or another one and should not add the upper bound.

But:
The current behavior is correct because is says **large-only** and by default the distributed files from the releases only generate the classes up to large.

When you clone the repository or https://github.com/zurb/foundation-zurb-template and add the needed classes to `$breakpoint-classes` it will also generate the needed breakpoints.

We could improve and check the current list in a future release to improve the current behavior.

But keep in your kind, bigger than large is still show-for-large as the breakpoints are mobile first and always using min-width. Which does exactly what you show.

If you want to generate the missing classes, you can create a project with https://github.com/zurb/foundation-zurb-template and adjust the settings.scss file.

**Comment by nagylzs** [comment]
All right than, I think we should close this issue. Thank you four your help!

**Comment by DanielRuf** [comment]
Also as a final information, we use `$breakpoints` in the mixin so developers who use SCSS can use the complete list.

So in your case it should use `$breakpoint-classes` but directly changing it there is not an option because this would break things for the developers who use SCSS. And this would be a big breaking change. We would need some other way or some alternative parameter which is used to tell the code that it should use `$breakpoint-classes` instead of `$breakpoints` for the generation of the dist files and remove the upper bound but this would not be what **only** really means.

So there are a few issues at the moment.

https://github.com/zurb/foundation-sites/blob/develop/scss/components/_visibility.scss#L22
